src/Response/ResponseExtensions.cs

```diff
@@ -58,5 +63,118 @@ public static class ResponseExtensions
 
             await negotiator.Handle(response.HttpContext.Request, response, obj, cancellationToken);
         }
+
+        /// <summary>
+        /// Copy a stream into the response body
+        /// </summary>
+        /// <param name="response">Current <see cref="HttpResponse"/></param>
+        /// <param name="stream">The <see cref="Stream"/> to copy from</param>
+        /// <param name="contentType">The content type for the response</param>
+        /// <param name="contentDisposition">The content disposition to allow file downloads</param>
+        /// <returns><<see cref="Task"/>/returns>
+        public static async Task FromStream(this HttpResponse response, Stream source, string contentType, ContentDisposition contentDisposition = null)
+        {
+            long rangeStart;
+            long rangeEnd;
+            var contentLength = source.Length;
+
+            response.Headers["AcceptRanges"] = "bytes";
+            response.ContentType = contentType;
+
+            if (contentDisposition != null)
+            {
+                response.Headers["Content-Disposition"] = contentDisposition.ToString();
+            }
+
+            // rangeHeader should be of the format "bytes=0-" or "bytes=0-12345" or "bytes=123-456"
+            var rangeHeader = response.HttpContext.Request.Headers["Range"].FirstOrDefault();
+
+            if (!string.IsNullOrEmpty(rangeHeader) && rangeHeader.Contains("="))
+            {
+                var rangeParts = rangeHeader.SplitOnFirst("=")[1].SplitOnFirst("-");
+                rangeStart = long.Parse(rangeParts[0]);
+                rangeEnd = rangeParts.Length == 2 && !string.IsNullOrEmpty(rangeParts[1])
+                    ? int.Parse(rangeParts[1]) // the client requested a chunk
```

Comment:
`rangeParts[0]` is parsed as `long`, while `rangeParts[1]` is parsed as `int`. Also, maybe use `TryParse` given that the `rangeHeader` format is not strictly validated.